Check qualified type for root stackness

Frassle opened the pull request:
With the introduction of component programs there will be mulitple "pulumi:pulumi:stack" resources in a program. We should therefore check against the qualified type of resources to see if they are the root stack, not just their direct type.

danielrbradley commented:
Oh, and for consistency, we might want to update `pkg/resource/edit/operations.go:L169` to use the `resource.RootStackType` too.

Frassle commented:
> The only one that I had in addition was in pkg/importer/hcl2.go line 131.

Fixed

> Oh, and for consistency, we might want to update pkg/resource/edit/operations.go:L169 to use the resource.RootStackType too.

Fixed

> What was your rational for why it's okay to completely remove the check for the root resource in the checkParent?

That check isn't the right check anymore. It used to be if trying to register a resource of type stack then the parent had to be empty. But for substacks they are of type stack but it's fine for them to have a parent. We can't actually tell from just a RegisterResource call if this is a root stack or not. We'd instead have to be like "is this the first stack we've seen, it must be the root". But that's fine to be a problem to solve later.
